Sources/Raygun_KSCrash/Recording/Raygun_KSCrashC.c

```diff
@@ -0,0 +1,301 @@
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+
+#include "Raygun_KSCrashC.h"
+
+#include "Raygun_KSCrashCachedData.h"
+#include "Raygun_KSCrashReport.h"
+#include "Raygun_KSCrashReportFixer.h"
+#include "Raygun_KSCrashReportStore.h"
+#include "Raygun_KSCrashMonitor_Deadlock.h"
+#include "Raygun_KSCrashMonitor_User.h"
+#include "Raygun_KSFileUtils.h"
+#include "Raygun_KSObjC.h"
+#include "Raygun_KSString.h"
+#include "Raygun_KSCrashMonitor_System.h"
+#include "Raygun_KSCrashMonitor_Zombie.h"
+#include "Raygun_KSCrashMonitor_AppState.h"
+#include "Raygun_KSCrashMonitorContext.h"
+#include "Raygun_KSSystemCapabilities.h"
+
+//#define Raygun_KSLogger_LocalLevel TRACE
+#include "Raygun_KSLogger.h"
+
+#include <inttypes.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+
+// ============================================================================
+#pragma mark - Globals -
+// ============================================================================
+
+/** True if KSCrash has been installed. */
+static volatile bool g_installed = 0;
+
+static bool g_shouldAddConsoleLogToReport = false;
+static bool g_shouldPrintPreviousLog = false;
+static char g_consoleLogPath[RAYGUN_KSFU_MAX_PATH_LENGTH];
+static Raygun_KSCrashMonitorType g_monitoring = Raygun_KSCrashMonitorTypeProductionSafeMinimal;
+static char g_lastCrashReportFilePath[RAYGUN_KSFU_MAX_PATH_LENGTH];
+
+
+// ============================================================================
+#pragma mark - Utility -
+// ============================================================================
+
+static void printPreviousLog(const char* filePath)
+{
+    char* data;
+    int length;
+    if(raygun_ksfu_readEntireFile(filePath, &data, &length, 0))
+    {
+        printf("\nvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv Previous Log vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv\n\n");
+        printf("%s\n", data);
+        printf("^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n");
+        fflush(stdout);
+    }
+}
+
+
+// ============================================================================
+#pragma mark - Callbacks -
+// ============================================================================
+
+/** Called when a crash occurs.
+ *
+ * This function gets passed as a callback to a crash handler.
+ */
+static void onCrash(struct Raygun_KSCrash_MonitorContext* monitorContext)
+{
+    if (monitorContext->currentSnapshotUserReported == false) {
+        RAYGUN_KSLOG_ERROR("Updating application state to note crash.");
```

Comment:
Debug changed to Error, looks unintentional here

Reply:
And 3 other places in this file

Reply:
Good catch I have fixed them up.